### PR TITLE
🐝 fix: supports legacy triggers

### PR DESCRIPTION
**Pull request by ptbrowne**
Claire has a legacy trigger whose content is encoded in base64. Banks did not recognize it as a konnector trigger and would display the "Install the connector" CTA, even if the konnector is already installed (and has a working trigger).

### CODE REVIEW

File: src/utils/triggers.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  if (trigger.message && trigger.message.konnector) {
+    return trigger.message.konnector
+  } else if (trigger.message.Data) {
+    // Legacy triggers
+    const message = JSON.parse(atob(trigger.message.Data))
+    return message.konnector
+  }
+}
+
+export { getKonnectorFromTrigger }
```
Comment (ptbrowne) [inline src/utils/triggers.js]: Do you think I should put this in cozy-doctypes/triggers ?

File: src/utils/triggers.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const getKonnectorFromTrigger = trigger => {
+  if (trigger.message && trigger.message.konnector) {
+    return trigger.message.konnector
+  } else if (trigger.message.Data) {
+    // Legacy triggers
+    const message = JSON.parse(atob(trigger.message.Data))
```
Comment (kosssi) [inline src/utils/triggers.js]: est ce que tu peux ajouter un log sentry pour que l'on puisse vérifier si on passe encore dans cette fonction dans 6 mois ?

File: src/utils/triggers.js (hunk 1 above)
Comment (drazik) [reply src/utils/triggers.js]: I don't think this is related to doctypes, so I would keep it on banks, at least for now.

File: src/utils/triggers.js (hunk 2 above)
Comment (ptbrowne) [reply src/utils/triggers.js]: Bonne idée

Comment (ptbrowne) [reply src/utils/triggers.js]: https://sentry.cozycloud.cc/sentry/cozy-banks-0s/issues/59669/

**Pull request merged**